src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,94 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { Filter } from '../../types/Filter';
+
+type Props = {
+  todos: Todo[],
+  filter: (filter: Filter) => void
+  removeCompletedTodos: () => void,
+  currentFilter: Filter
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filter,
+  removeCompletedTodos: removeAll,
+  currentFilter,
+}) => {
+  const loadFilteringTodos = (isActive: boolean) => {
+    if (!isActive) {
+      filter('active');
+    } else {
+      filter('completed');
+    }
+  };
+
+  const allTodo = () => {
+    filter('all');
+  };
+
+  const isCompletedTodos = todos.some(todo => todo.completed);
+  const activeTodosCount = (
+    todos.length - todos.filter(
+      item => item.completed,
+    ).length);
+  const removeCompletedTodos = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    removeAll();
+
+    return null;
+  };
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
```

Comment:
Consider adding a more descriptive class name for better readability and maintainability. 📝